src/mapping/AxialGeoMultiscaleMapping.cpp

```diff
@@ -0,0 +1,177 @@
+  const int outValueDimensions = outData.size() / output()->vertices().size();
+
+  int effectiveCoordinate = static_cast<std::underlying_type_t<MultiscaleType>>(_axis);
+  PRECICE_ASSERT(effectiveCoordinate == static_cast<std::underlying_type_t<MultiscaleType>>(MultiscaleAxis::X) ||
+                     effectiveCoordinate == static_cast<std::underlying_type_t<MultiscaleType>>(MultiscaleAxis::Y) ||
+                     effectiveCoordinate == static_cast<std::underlying_type_t<MultiscaleType>>(MultiscaleAxis::Z),
+                 "Unknown multiscale axis type.")
+
+  PRECICE_ASSERT((inputValues.size() / inValueDimensions == static_cast<int>(input()->vertices().size())),
+                 inputValues.size(), inValueDimensions, input()->vertices().size());
+  PRECICE_ASSERT((outputValues.size() / outValueDimensions == static_cast<int>(output()->vertices().size())),
+                 outputValues.size(), outValueDimensions, output()->vertices().size());
+
+  PRECICE_DEBUG("Map consistent");
+  if (_type == MultiscaleType::SPREAD) {
+    /*
+      3D vertices are assigned a value based on distance from the 1D vertex.
+      Currently, a Hagen-Poiseuille profile determines the velocity value.
+    */
+    PRECICE_ASSERT(input()->vertices().size() == 1);
+    size_t const outSize = output()->vertices().size();
+
+    for (size_t i = 0; i < outSize; i++) {
+      PRECICE_ASSERT(static_cast<int>((i * outValueDimensions) + effectiveCoordinate) < outputValues.size(), ((i * outValueDimensions) + effectiveCoordinate), outputValues.size());
+      // When adding support for 2D, remember that this should be 1.5 * inputValues(effectiveCoordinate) * (1 - (_vertexDistances[i] * _vertexDistances[i]));
+      outputValues((i * outValueDimensions) + effectiveCoordinate) = 2 * inputValues(effectiveCoordinate) * (1 - (_vertexDistances[i] * _vertexDistances[i]));
+    }
+  } else {
+    PRECICE_ASSERT(_type == MultiscaleType::COLLECT);
+    /*
+      1D vertex is assigned the averaged value over all 3D vertices at the outlet,
+      but only of the effectiveCoordinate component of the velocity vector.
+    */
+    PRECICE_ASSERT(output()->vertices().size() == 1);
+    outputValues(effectiveCoordinate) = 0;
+    size_t const inSize               = input()->vertices().size();
+    for (size_t i = 0; i < inSize; i++) {
+      PRECICE_ASSERT(static_cast<int>((i * inValueDimensions) + effectiveCoordinate) < inputValues.size(), ((i * inValueDimensions) + effectiveCoordinate), inputValues.size())
+      outputValues(effectiveCoordinate) += inputValues((i * inValueDimensions) + effectiveCoordinate);
+    }
+    outputValues(effectiveCoordinate) = outputValues(effectiveCoordinate) / inSize;
+  }
+}
+
+void AxialGeoMultiscaleMapping::tagMeshFirstRound()
+{
+  PRECICE_TRACE();
+
+  computeMapping();
+
+  if (getConstraint() == CONSISTENT) {
+    PRECICE_ASSERT(_type == MultiscaleType::SPREAD, "Not yet implemented");
+    PRECICE_ASSERT(input()->vertices().size() == 1);
```

Comment:
I do not understand the "not yet implemented" here.

Codecov complains that we don't cover this with tests, but we do have axial consistent collect in the integration tests:

```xml
    <mapping:axial-geometric-multiscale
      direction="read"
      multiscale-type="spread"
      multiscale-radius="1.0"
      multiscale-axis="z"
      from="Mesh1D"
      to="Mesh3D"
      constraint="consistent" />
```

We also have the respective implementation inside `mapConsistent`.

Does this mean that this code is generally unreachable?

Reply:
We don't allow partitioned meshes at the moment, so this could be dropped.

Reply:
Leaving this open until the rest of the tests succeed again and I get the actual test coverage.

Reply:
The test coverage now seems to be fine. As far as I understand, we only miss parts that we actually have not yet implemented yet.

@fsimonis please have a look at the CodeCov output. Is there anything we can improve there?

Reply:
Mesh tagging is only used when meshes are partitioned aka the solver runs in parallel, which isn't the case here.